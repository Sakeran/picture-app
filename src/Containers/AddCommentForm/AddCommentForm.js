import React from 'react';
import { graphql, gql } from 'react-apollo';

import { commentsQuery } from '../CommentListContainer/CommentListContainer';

import './AddCommentForm.css';

class AddCommentForm extends React.Component {
  constructor() {
    super();
    this.state = {
      commentText: ''
    }
  }

  handleChange = (e) => {
    this.setState({
      commentText: e.target.value
    });
  }

  addComment = (e) => {
    e.preventDefault();
    if (!this.state.commentText) {
      return;
    }
    this.props.mutate({
      variables: {
        postId: this.props.postId,
        message: this.state.commentText
      },
      update: (store, { data: { addComment } }) => {
        if (!addComment) { return; }
        const data = store.readQuery({
          query: commentsQuery,
          variables: {
            postId: this.props.postId,
            offset: 0
          }
        });
        data.comments.unshift(addComment);
        store.writeQuery({
          query: commentsQuery,
          variables: {
            postId: this.props.postId,
            offset: 0
          },
          data
        });
      }
    })
    .then(res => {
      this.setState({
        commentText: ''
      });
    });
  }

  render() {
    return (
      <div id="addComment" className="AddCommentForm">
        <h3 className="centered">Add a new comment:</h3>
        <form onSubmit={this.addComment}>
          <div className="field">
            <textarea name="comment"
              placeholder="Enter a comment here."
              value={this.state.commentText}
              onChange={this.handleChange}
            />
          </div>
          <div className="field">
            <input className="button" type="submit" value="Add Comment" />
          </div>
        </form>
      </div>
    )
  }
};

const addCommentMutation = gql`
  mutation addComment($postId: ID!, $message: String!) {
    addComment(postId: $postId, message: $message) {
      id
      text
      date
      user {
        id
        username
      }
    }
  }
`;

export { AddCommentForm };
export default graphql(addCommentMutation)(AddCommentForm);
