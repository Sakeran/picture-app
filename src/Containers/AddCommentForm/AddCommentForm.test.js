import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AddCommentForm } from './AddCommentForm';

test('AddCommentForm renders correctly', () => {
  const component = shallow(<AddCommentForm />);
  expect(component.find('textarea[name="comment"]')).toHaveLength(1);
  expect(component.find('input[type="submit"]')).toHaveLength(1);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
