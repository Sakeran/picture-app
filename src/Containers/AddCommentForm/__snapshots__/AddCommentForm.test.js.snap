// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCommentForm renders correctly 1`] = `
<div
  className="AddCommentForm"
  id="addComment"
>
  <h3
    className="centered"
  >
    Add a new comment:
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <textarea
        name="comment"
        onChange={[Function]}
        placeholder="Enter a comment here."
        value=""
      />
    </div>
    <div
      className="field"
    >
      <input
        className="button"
        type="submit"
        value="Add Comment"
      />
    </div>
  </form>
</div>
`;
